# HVAC Linear Algebra project

![HVAC diagram showing the flow rates and connections between floors](../images/hvac.png)

In the figure above you have an idealized Heating, Ventilation and Air conditioning (HVAC) system. In the current configuration, the three-room building is being cooled off by $T_{in}=15^oC$ air fed into the building at 0.1 kg/s. Our goal is to determine the steady-state temperatures of the rooms given the following information

* $\dot{m}_1=0.1~kg/s$
* $\dot{m}_2=0.15~kg/s$
* $\dot{m}_3=0.17~kg/s$
* $\dot{m}_4=0.1~kg/s$
* $\dot{m}_5=0.07~kg/s$
* $\dot{m}_6=0.05~kg/s$
* $C_p=1000~\frac{J}{kg-C}$
* $\dot{Q}_{in} = 500~W$
* $T_{in} = 15^{o} C$

The energy-balance equations for rooms 1-3 create three equations:

1. $\dot{m}_1 C_p T_{in}+\dot{Q}_{in}-\dot{m}_2 C_p T_{1}+\dot{m}_6 C_p T_{2} = 0$

2. $\dot{m}_2 C_p T_{1}+\dot{Q}_{in}+\dot{m}_5 C_p T_{3}-\dot{m}_3 C_p T_{2}-\dot{m}_6 C_p T_{2} = 0$

3. $\dot{m}_3 C_p T_{2}+\dot{Q}_{in}-\dot{m}_5 C_p T_{3}-\dot{m}_4 C_p T_{3} = 0$

Identify the unknown variables and constants to create a linear algebra problem in the form of $\mathbf{Ax}=\mathbf{b}$.

a. Create the matrix $\mathbf{A}$

b. Create the known vector $\mathbf{b}$

c. Solve for the unknown variables, $\mathbf{x}$

d. What are the warmest and coldest rooms? What are their temperatures?

In [1]:
import numpy as np

In [2]:
Q = 500
m1 = 0.1
m2 = 0.15
m3 = 0.17
m4 = 0.1
m5 = 0.07
m6 = 0.05
C = 1000
Tin = 15
# part (a)

A = np.array([[-m2*C, m6*C, 0],
              [m2*C, -(m3+m6)*C, m5*C],
              [0, m3*C, -(m4+m5)*C]])
print(A)

[[-150.   50.    0.]
 [ 150. -220.   70.]
 [   0.  170. -170.]]


In [3]:
# part (b)
b = np.array([-Q-m1*C*Tin,
             -Q,
             -Q])
print(b)

[-2000.  -500.  -500.]


In [4]:
# part (c)
x = np.linalg.solve(A,b)
print(x)

[22.35294118 27.05882353 30.        ]


# Part (d) solution:

The warmest room is number 3 - $T_3 = 30^o$

The coolest room is number 1 - $T_1 = 22.4^o$